### PR TITLE
chore: add vscode recommendations; jira commit tagging

### PR DESCRIPTION
**Pull request by castastrophe**
## Description

Add a few VSCode suggestions to help streamline onboarding for new users - I can record a walk-through of how these can facilitate productivity as well. I also included the requisite workspace settings to make setting up these utilities easier.

Added some logic to postfix JIRA information (if it exists in the branch name) to commit messages.

## To-do list
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [x] This pull request is ready to merge.

### CODE REVIEW

File: .husky/prepare-commit-msg [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# git prepare-commit-msg hook for automatically prepending an issue key
+# from the start of the current branch name to commit messages.
+
+# check if commit is merge commit or a commit ammend
+if [ $2 = "merge" ] || [ $2 = "commit" ]; then
+    exit
+fi
+
+ISSUE_KEY=`git branch | grep -o "\* \(.*/\)*CSS-[0-9]\+" | grep -o "CSS-[0-9]\+"`
+if [ $? -ne 0 ]; then
```
Comment (castastrophe) [inline .husky/prepare-commit-msg:12]:
```suggestion
# If the parse above returns the full branch name, default to the original message
if [ "$ISSUE_KEY" == "$( git rev-parse --abbrev-ref HEAD)" ]; then
```

**Review by castastrophe — COMMENTED** [review]
A few possible enhancements

File: .husky/prepare-commit-msg [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ISSUE_KEY=`git rev-parse --abbrev-ref HEAD | sed -r 's/^.*[CSS|JIRA][-|_]{0,1}([0-9]+).*$/CSS-\1/I'`
+
+# If the parse above returns the full branch name, default to the original message
+# Must quote variable and the command to compare strings in bash
+[[ "$ISSUE_KEY" == "$(git rev-parse --abbrev-ref HEAD)" ]] && exit
+
+# issue key matched from branch prefix, postfix to commit message
+TEMP=`mktemp /tmp/commitmsg-XXXXX`
+(echo "$(cat  $1) [$ISSUE_KEY]") > $TEMP
+cat $TEMP > $1
```
Comment (jawinn) [inline .husky/prepare-commit-msg:19]: I gave this a try with a branch named `css-500-test` and `CSS-500-test2` but did not see it adding the ticket number to the commit messages. I made a commit on each and then looked at the `git log` and did not see any changes to it.

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,3 +1,104 @@
...
+	},
+	"emmet.syntaxProfiles": {
+		"postcss": "css"
+	},
+	"eslint.packageManager": "yarn",
+	"files.associations": {
+		"*.css": "postcss"
+	},
+	"githubIssues.issueBranchTitle": "${author}/${sanitizedIssueTitle}-gh-${issueNumber}",
+	"githubIssues.queries": [
```
Comment (jawinn) [inline .vscode/settings.json:63]: 👍 Appears to be working. I installed the extension and saw many issues populated when starting to type `#` in a comment to reference an issue.

File: .vscode/settings.json [hunk 4]
```diff
@@ -1,3 +1,104 @@
...
+		{
+			"label": "Created issues",
+			"query": "author:${user} state:open repo:${owner}/${repository} sort:created-desc"
+		},
+		{
+			"label": "Recent issues",
+			"query": "state:open repo:${owner}/${repository} sort:updated-desc"
+		}
+	],
+	"githubPullRequests.queries": [
```
Comment (jawinn) [inline .vscode/settings.json:77]: 👍 Looks good as well. I see these categories appearing with results when I view "GitHub" in the VS Code sidebar.

**Review by jawinn — CHANGES_REQUESTED** [review]

**Comment by castastrophe** [comment]
@jawinn I removed the commit message parsing tasks so we can merge this in with just the extensions recommendations 🥳

**Pull request merged**